### PR TITLE
feat: multi-threading for viability phase

### LINKED ISSUE #89: Multi-threading for viability phase

### PR DESCRIPTION
**Pull request by sporreking**
Closes #89

### CODE REVIEW

**Comment by sporreking** [comment]
To test code, copy _mock-init_ section at bottom of file from [here](https://github.com/MattiasBeming/LiU-AI-Project-Active-Learning-for-Music/blob/viability_phase_mock_init/src/viability_phase.py) and delete invalid/undesired datasets. It is probably desirable to remove `max_std_sampling`, and `ds3` dataset from `init()`.

File: src/viability_phase.py [hunk 1]
```diff
@@ -17,19 +90,49 @@ def viability_phase(learning_profiles: list, num_iterations: int = -1,
...
+        th.start()
+        threads.append(th)
+
+    # Wait for threads to exit
+    for th in threads:
+        th.join()
+
+    tqdm.write("All threads finished successfully!")
 
         yield lp
```
Comment (MattiasBeming) [inline src/viability_phase.py:137]: How did the code run with indentation error? 

The yielding of each profile after they have completed does not work now.
This was implemented like this bc `evaluation_phase` can save (to disk) the profile after it has completed, and not wait for all profiles to finish before saving.

**Review by MattiasBeming — CHANGES_REQUESTED** [review]

File: src/viability_phase.py (hunk 1 above)
Comment (sporreking) [reply src/viability_phase.py:137]: Definitely an error, good catch!

**Review by MattiasBeming — APPROVED** [review]
Good job

**Pull request merged**